code/hdb/hdbstandard.q

```diff
@@ -4,4 +4,28 @@ reload:{
 	system"l ."}
 
 // Get the relevant HDB attributes
-.proc.getattributes:{`date`tables!(@[value;`date;`date$()];tables[])}
+\d .proc
+getattributes:{
+	timecolumns:1!except[(::),gettimecolumns each t:tables[`.];(::)];
+	attrtable:`date`tables`procname!(
+		/ date reporting makes some assumptions
+		@[value;`date;`date$()] union first[d]+til 1+last deltas d:exec(min;max)@\:distinct`date$last each raze value each timecolumns from timecolumns;
+		tables[];
+		.proc.procname);
+	attrtable[`dataaccess]:`segid`tablename!(first .ds.segmentid;(key timecolumns)!select wc,timecolumns from update wc:{""}'[i],timecolumns from timecolumns);
+    	attrtable}
+// Get temporal cols and the time windows they contain
+gettimecolumns:{
+	tcols:exec c from meta t:$[`date in cols x;
+    	/ select table using max date if date col exists
+	select from x where date=max date;
+                value x] where t in "pdz";
+	if[not count tcols;:(::)];
```

Comment:
do you need to return a generic null? more common would be to return an empty list i.e.

```suggestion
	if[not count tcols;:()];
```